### PR TITLE
UHM-4830, add data entry validations

### PR DESCRIPTION
**Pull request by cioan**
Here is the implementation for the data validations outlined in https://pihemr.atlassian.net/browse/UHM-4830. Thanks! @mseaton , @mogoodrich , @brandones

### CODE REVIEW

File: configuration/pih/htmlforms/hiv/zl/hiv-dispensing.xml [hunk 1]
```diff
@@ -105,22 +105,143 @@
...
+            getField('chwName.value').prop( 'disabled', true );
+        } else if (dispensedToValue == dispensedToChwConceptId) {
+            getField('dispensedInCommunity.value').prop( 'disabled', true );
+        }
+
+        var validateNextDispensingDate = function() {
+            var today = new Date();
+            var currentEncounterDate = '<lookup expression="encounter.getEncounterDatetime().getTime()"/>';
+            if (typeof currentEncounterDate !== "undefined" &amp;&amp; currentEncounterDate !== null &amp;&amp; (currentEncounterDate.length > 0)) {
+              today = new Date(+currentEncounterDate);
```
Comment (mogoodrich) [inline configuration/pih/htmlforms/hiv/zl/hiv-dispensing.xml:124]: Should this variable be renamed to currentEncounterDate?  If the encounter is in the past, then "today" is no longer "today"?

File: configuration/pih/htmlforms/hiv/zl/hiv-dispensing.xml [hunk 2]
```diff
@@ -105,22 +105,143 @@
...
+                getField('nextDispenseDate.error').text("<uimessage code='Next dispensing date has to be within a year of the encounter date'/>").show();
+                return false;
+            } else {
+                getField('nextDispenseDate.error').text("").hide();
+                return true;
+            }
+        };
+
+        jq("#medsTable select").on('change', function() {
+          if( this.value.length > 0 );
```
Comment (mogoodrich) [inline configuration/pih/htmlforms/hiv/zl/hiv-dispensing.xml:143]: This just ends in a semicolon, does this do anything?

File: configuration/pih/htmlforms/hiv/zl/hiv-dispensing.xml [hunk 3]
```diff
@@ -105,22 +105,143 @@
...
+        });
 
+
+        jq("#monthsDispensed input").change(function() {
+          var monthsDispensed= getValue('monthsDispensed.value');
+          if (typeof monthsDispensed !== "undefined" &amp;&amp; monthsDispensed !== null &amp;&amp; (monthsDispensed.length > 0)) {
               var today = new Date();
               var currentEncounterDate = '<lookup expression="encounter.getEncounterDatetime().getTime()"/>';
               if (typeof currentEncounterDate !== "undefined" &amp;&amp; currentEncounterDate !== null &amp;&amp; (currentEncounterDate.length > 0)) {
                 today = new Date(+currentEncounterDate);
```
Comment (mogoodrich) [inline configuration/pih/htmlforms/hiv/zl/hiv-dispensing.xml:195]: Same comment about "today"

**Review by mogoodrich — COMMENTED** [review]
Thanks @cioan !  Added a few comments.  FWIW, I just looked at the code itself, I didn't cross-reference with the business validation rules, I'll leave that for testing... :)

**Comment by cioan** [comment]
Thanks @mogoodrich ! I have corrected the items you pointed out. Thanks!

**Review by mogoodrich — APPROVED** [review]
Thanks @cioan !

**Pull request merged**